pkg/receiver/gitproviderreceiver/internal/scraper/gitlabscraper/gitlab_scraper.go

```diff
@@ -70,56 +102,66 @@ func (gls *gitlabScraper) scrape(ctx context.Context) (pmetric.Metrics, error) {
 
 	graphClient := graphql.NewClient("https://gitlab.com/api/graphql", gls.client)
 
-	subgroups, err := getGroupDescendents(context.Background(), graphClient, gls.cfg.GitLabOrg)
-
-	var projectList []string
+	var projectList []gitlabProject
+	var projectsCursor *string
 
-	if len(subgroups.Group.DescendantGroups.Nodes) > 0 {
-		for _, d := range subgroups.Group.DescendantGroups.Nodes {
-			projectList = append(projectList, d.Name)
-		}
-	}
-
-	// TODO: this only works for top level groups with 1 level of subgroups
-	for _, group := range subgroups.Group.DescendantGroups.Nodes {
-		// full path should be a function that takes in a list of strings and returns a string
-		// concatenated together. This makes it testable & works with multiple subgroups
-		projects, err := getGroupProjects(context.Background(), graphClient, gls.cfg.GitLabOrg+"/"+group.Name)
+	for hasNextPage := true; hasNextPage; {
+		// Get the next page of data
+		projects, err := getAllGroupProjects(context.Background(), graphClient, gls.cfg.GitLabOrg, projectsCursor)
 		if err != nil {
-			gls.logger.Sugar().Errorf("error getting descendant groups: %v", err)
+			gls.logger.Sugar().Errorf("error: %v", err)
 		}
-		for _, project := range projects.Group.Projects.Nodes {
-			projectList = append(projectList, project.Name)
+
+		if len(projects.Group.Projects.Nodes) == 0 {
+			err = errors.New("no GitLab projects found for the given group/org")
+			gls.logger.Sugar().Error(err)
+
+			return gls.mb.Emit(), err
 		}
-	}
 
-	// TODO: Must paginate and update query to get all branches of projects in the list
-	//for _, project := range projectList {
+		// Check if there is a next page
+		hasNextPage = projects.Group.Projects.PageInfo.HasNextPage
 
-	//	gls.logger.Sugar().Debugf("project: %v", project)
+		// Set the cursor to the end cursor of the current page
+		projectsCursor = &projects.Group.Projects.PageInfo.EndCursor
 
-	//	branches, err := getBranchNames(context.Background(), graphClient, project)
+		for _, p := range projects.Group.Projects.Nodes {
+			projectList = append(projectList, gitlabProject{
+				Name:           p.Name,
+				Path:           p.FullPath,
+				CreatedAt:      p.CreatedAt,
+				LastActivityAt: p.LastActivityAt,
+			})
+		}
+	}
 
-	//	if err != nil {
-	//		gls.logger.Sugar().Errorf("error getting branches: %v", err)
-	//	}
+	ch := make(chan branchData)
+	var wg sync.WaitGroup
 
-	//	var branchList []string
-	//	branchList = append(branchList, branches.Project.Repository.BranchNames...)
+	for _, project := range projectList {
+		wg.Add(1)
 
-	//	gls.logger.Sugar().Debugf("branch: %v", branches.Project.Repository.BranchNames)
+		// TODO: Must account for when there are more than 100,000 branch names in a project.
+		go getBranches(ctx, graphClient, project.Path, ch, &wg)
```

Comment:
This is the offending line complained about by the linter. `getBranches` returns `err` but it's not being checked.